platforms/core-runtime/messaging/src/main/java/org/gradle/internal/event/AnonymousListenerBroadcast.java

```diff
@@ -29,7 +29,7 @@ public AnonymousListenerBroadcast(Class<T> type, Dispatch<MethodInvocation> forw
     }
 
     @Override
-    public void removeAll() {
+    public synchronized void removeAll() {
         super.removeAll();
         add(forwardingDispatch);
```

Comment:
:x: as we don't use `synchronized` when reading the `broadcast`, we may see it between `removeAll` and `add(forwardingDispatch)`, so we may miss broadcasting to `forwardingDispatch`.

Reply:
Introduced a method in the base class to replace `broadcast` atomically